cpp/src/rolling/rolling_detail.cuh

```diff
@@ -895,6 +899,235 @@ struct rolling_window_launcher {
       stream,
       mr);
   }
+
+  template <typename PrecedingIter, typename FollowingIter>
+  std::unique_ptr<column> get_collect_list_offsets(column_view const& input,
+                                                   PrecedingIter preceding_begin,
+                                                   FollowingIter following_begin,
+                                                   size_type min_periods,
+                                                   rmm::cuda_stream_view stream,
+                                                   rmm::mr::device_memory_resource* mr)
+  {
+    using namespace cudf;
+
+    // Materialize offsets column.
+    auto size_data_type = data_type{type_to_id<size_type>()};
+    auto sizes =
+      make_fixed_width_column(size_data_type, input.size(), mask_state::UNALLOCATED, stream, mr);
+    auto mutable_sizes = sizes->mutable_view();
+    thrust::transform(thrust::device,
+                      preceding_begin,
+                      preceding_begin + input.size(),
+                      following_begin,
+                      mutable_sizes.begin<size_type>(),
+                      [min_periods] __device__(auto preceding, auto following) {
+                        return (preceding + following) < min_periods ? 0 : (preceding + following);
+                      });
+    return strings::detail::make_offsets_child_column(
+      sizes->view().begin<size_type>(), sizes->view().end<size_type>(), stream, mr);
+  }
+
+  template <typename PrecedingIter, typename FollowingIter>
```

Comment:
Missing doc

Reply:
These helper functions are private to the implementation of `COLLECT`. I will add `@brief`s and descriptions to explain them.